src/workerd/jsg/modules.c++

```diff
@@ -9,8 +9,47 @@
 #include <set>
 
 namespace workerd::jsg {
+
 namespace {
+// This list must be kept in sync with the list of builtins from Node.js.
+// It should be unlikely that anything is ever removed from this list, and
+// adding items to it is considered a semver-major change in Node.js.
+static const std::set<kj::StringPtr> NODEJS_BUILTINS {
+  "_http_agent",         "_http_client",        "_http_common",
+  "_http_incoming",      "_http_outgoing",      "_http_server",
+  "_stream_duplex",      "_stream_passthrough", "_stream_readable",
+  "_stream_transform",   "_stream_wrap",        "_stream_writable",
+  "_tls_common",         "_tls_wrap",           "assert",
+  "assert/strict",       "async_hooks",         "buffer",
+  "child_process",       "cluster",             "console",
+  "constants",           "crypto",              "dgram",
+  "diagnostics_channel", "dns",                 "dns/promises",
+  "domain",              "events",              "fs",
+  "fs/promises",         "http",                "http2",
+  "https",               "inspector",           "inspector/promises",
+  "module",              "net",                 "os",
+  "path",                "path/posix",          "path/win32",
+  "perf_hooks",          "process",             "punycode",
+  "querystring",         "readline",            "readline/promises",
+  "repl",                "stream",              "stream/consumers",
+  "stream/promises",     "stream/web",          "string_decoder",
+  "sys",                 "timers",              "timers/promises",
+  "tls",                 "trace_events",        "tty",
+  "url",                 "util",                "util/types",
+  "v8",                  "vm",                  "worker_threads",
+  "zlib"
+};
+
+}  // namespace
+
+void logIfNodeSpecifier(kj::StringPtr specifier) {
+  if (NODEJS_BUILTINS.contains(specifier)) {
+    LOG_NOSENTRY(WARNING, "Using require() to import a module with a known Node.js built-in name",
```

Comment:
Won't this be very noisy? Should this be a periodic log? or possibly a metric in prometheus?

Reply:
Part of the idea here is to get a sense of just how often folks are using modules that use the same name as Node.js builtins. Because of the way module imports work, it should happen at most once per isolate, not once per request unless they are using `require(...)` which also gives us good information. The intent is to keep this in for only about two weeks then revert it back out. Also the `NOSENTRY` should limit noise impact.